packages/dtslint-runner/src/post-results.ts

```diff
@@ -0,0 +1,151 @@
+
+  try {
+    let newComment;
+    if (status === "fail") {
+      newComment = `Hey @${userToTag}, it looks like the DT test run failed. Please check the log for more details. ${checkLogsMessage}`;
+    }
+    else {
+      const nightlyErrors: Errors = [];
+      if (nightlyErrorsPath) {
+        const nightlyFiles = glob.sync(`**/*.json`, { cwd: nightlyErrorsPath, absolute: true });
+        for (const file in nightlyFiles) {
+          nightlyErrors.push(...JSON.parse(readFileSync(file, "utf-8")) as Errors);
+        }
+      }
+      const branchErrors: Errors = [];
+      if (branchErrorsPath) {
+        const branchFiles = glob.sync(`**/*.json`, { cwd: branchErrorsPath, absolute: true });
+        for (const file in branchFiles) {
+          branchErrors.push(...JSON.parse(readFileSync(file, "utf-8")) as Errors);
+        }
+      }
+
+      newComment = `Hey @${userToTag}, the results of running the DT tests are ready.\n`;
+      const diffComment = getDiffComment(nightlyErrors, branchErrors);
+      if (diffComment) {
+        newComment += `There were interesting changes:`
+        if (newComment.length + diffComment.length + checkLogsMessage.length > 65535) {
+          newComment += `Changes are too big to display here, please check the log.`
+        }
+        else {
+          newComment += diffComment;
+        }
+      }
+      else {
+        newComment += "Everything looks the same!"
+      }
+      newComment += checkLogsMessage;
+    }
+    
+    const response = await gh.issues.createComment({
+      issue_number: +issue,
+      owner: "Microsoft",
+      repo: "TypeScript",
+      body: newComment
+    });
+
+    const newCommentUrl = response.data.html_url;
+    const statusComment = await gh.issues.getComment({
+      owner: "Microsoft",
+      repo: "TypeScript",
+      comment_id: +statusCommentId,
+    });
+
+    const newBody = `${statusComment.data.body}\n\nUpdate: [The results are in!](${newCommentUrl})`;
+    await gh.issues.updateComment({
+      owner: "Microsoft",
+      repo: "TypeScript",
+      comment_id: +statusCommentId,
+      body: newBody
+    });
+  }
+  catch (e) {
+    await gh.issues.createComment({
+      issue_number: +issue,
+      owner: "Microsoft",
+      repo: "TypeScript",
+      body: `Hey @${userToTag}, something went wrong when publishing results. ${checkLogsMessage}`
+    });
+  }
+}
+
+function getDiffComment(nightly: Errors, branch: Errors): string | undefined {
+  const nightlyMap = new Map(nightly.map(error => [error.path, error]));
+  const branchMap = new Map(branch.map(error => [error.path, error]));
+
+  const nightlyOnly = [];
+  const bothChanged = [];
+
+  for (const [path, error] of nightlyMap) {
+    if (branchMap.has(path)) {
+      const branchError = branchMap.get(path)!;
+      if (branchError.message !== error.message) {
```

Comment:
How much noise does this suppress? I assume that it's better than `diff` that we're currently using.